lib/functions/installs.bash

```diff
@@ -253,3 +422,57 @@ install_tool_version() {
     fi
   fi
 }
+
+get_legacy_version_in_dir() {
+  local plugin_name=$1
+  local search_path=$2
+  local legacy_filenames=$3
+
+  local asdf_version
+  for filename in $legacy_filenames; do
+    local legacy_version
+    legacy_version=$(parse_legacy_version_file "$search_path/$filename" "$plugin_name")
+
+    if [ -n "$legacy_version" ]; then
+      printf "%s\n" "$legacy_version"
+      return 0
+    fi
+  done
+}
+
+stringlist_a_subset_of_b() {
+  local list_a=$1
+  local list_b=$2
+  local array_a
+  IFS=' ' read -r -a array_a <<<"$list_a"
+  for item_a in "${array_a[@]}"; do
+    if [[ -z $(stringlist_contains "$list_b" "$item_a") ]]; then
+      return 0
+    fi
+  done
+  printf "true\n"
+  return 0
+}
+
+stringlist_contains() {
+  local list=$1
+  local search=$2
+  local array
+  IFS=' ' read -r -a array <<<"$list"
+  for item in "${array[@]}"; do
+    if [ $item = $search ]; then
+      printf "true\n"
+      return 0
+    fi
+  done
+}
+
+display_debug() {
+  if [[ $DEBUG = "true" ]]; then
+    printf "debug: %s\n" "$1" >&2
+  fi
+}
+
+display_none() {
+  printf ""
+}
```

Comment:
Might be good to put these in `utils.sh`.

Reply:
Done